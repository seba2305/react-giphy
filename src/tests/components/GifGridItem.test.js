import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://www.google.cl';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe mostrar el componente enviando params correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('debe tener animate__fadeInDown', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeInDown')).toBeTruthy();
        
    })
    

})
