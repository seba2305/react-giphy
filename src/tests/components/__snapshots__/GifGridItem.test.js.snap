// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente enviando params correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Un titulo"
    src="https://www.google.cl"
  />
  <p>
    Un titulo
  </p>
</div>
`;
